src/include/simeng/RegisterValue.hh

```diff
@@ -128,7 +128,7 @@ class RegisterValue {
 
   /** The underlying local member value. Aligned to 8 bytes to prevent
    * potential alignment issue when casting. */
-  alignas(8) char value[MAX_LOCAL_BYTES];
+  alignas(8) char localValue[MAX_LOCAL_BYTES] = {};
```

Comment:
This is slightly inefficient as it initialises localValue to all 0's for every instance which will be overwritten when constructing the instance. We should test if there is any visible performance degradation from this change. Maybe another one for Leo @ABenC377

Reply:
Alternative solution is to do something like 

```
#pragma GCC diagnostic push
#pragma GCC diagnostic ignored "-Wuninitialized"
    foo(b);         /* no diagnostic for this one */
#pragma GCC diagnostic pop
```

Reply:
Given we use Pool's for RegisterValues this _may_ not cause a huge performance hit. Worth checking though